tracer/src/Datadog.Trace/Debugger/Instrumentation/AsyncMethodDebuggerInvoker.SingleProbe.cs

```diff
@@ -234,6 +236,24 @@ public static void LogLocal<TLocal>(ref TLocal local, int index, ref AsyncMethod
             asyncState.SnapshotCreator.StartSampling();
         }
 
+        private static bool ProcessNotHoistedMoveNextLocal<TLocal>(ref TLocal local, AsyncMethodDebuggerState asyncState, string localName)
```

Comment:
I think you should also introduce this fix to `AsyncLineDebuggerInvoker` & possibly craft a test.

Reply:
This specific issue is only with async method probe and not hoisted local\s. We don't mean to cover all  other cases where local is out of scope or nulled before we captured it.